enumarable_methods.rb

```diff
@@ -0,0 +1,74 @@
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+      return to_enum unless block_given?
+    end
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    selection = []
+    my_each { |val| selection.push(val) if yield(val) }
+    selection
+  end
+end
+
+def my_all?(value = nil)
+  result = true
+  if value
+    my_each { |element| result &&= element == value }
+  else
+    my_each { |element| result &&= yield(element) }
+  end
+  result
+end
+
+def my_any?
+  my_each { |val| return true if yield(val) }
+  false
+end
+
+def my_none?
+  my_each { |val| return false if yield(val) }
+  true
+end
+
+def my_count
+  count = 0
+  my_each { |val| count += 1 if yield(val) }
+  count
+end
+
+def my_map(proc = nil)
+  result = []
+  my_each do |element|
+    result <<
+      if proc
+        proc.call(element)
+      else
+        yield(element)
+      end
+  end
+  result
+end
+
+def my_inject(initial = self[0])
+  result = initial
+  my_each { |val| result = yield(result, val) }
+  result
+end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should work with symbols as arguments (:+) and perform the right operation.